### ISSUE TITLE
exclude: dependency anaysis fail when using bundles files and text! plugin

### ISSUE THREAD

**Issue by DimitarChristoff** [jrburke/r.js#724]
This is a slight edge-case but it's also quite frustrating

Using multiple nested repos that try to build to local stuff and leaving others as excluded so the consumption app can add them just once.

so. say we have a bundled file for that has this:

``` javascript
define('text!src/main.html',[],function () { return 'I am a teapot\n';});
define('foo',['require'],function(require){ return 'foo';});
define('bar',['require'],function(require){ return function(tpl){console.log(tpl);};});
define('baz',['require','text!./main.html','./foo','./bar'],function(require){
        // main build target
    var template = require('text!./main.html'),
        foo = require('./foo'), // foo.js
        bar = require('./bar'); // bar.js
    return function(){
        bar(template);
    };
});
```

built with `exclude: ['text']` it is consumed by another app that wants to use it, build itself but not include the above.

So. 

``` javascript
require.config({
    bundles: {
        'dist/baz-min.js': ['baz','foo','bar']
    }
});
```

When being built by r.js, passing:

``` javascript
exclude: ['text', 'dist/baz-min.js']
include: ['ourownscript']
```

`ourownscript.js` requires and uses baz directly or any sub module and works fine in the browser. but the r.js on this produces:

```
Tracing dependencies for: dist/baz-min.js
{ [Error: Error: Loader plugin did not call the load callback in the build:
text:
  text!main.html
Module loading did not complete for: 
    at Function.build.checkForErrors (... node_modules\grunt-contrib-requirejs\node_modules\requirejs\bin\r.js:27417:19)
```

TL;DR: goal: produce minimal local builds that don't include other people's builds or vendor plugins--analysys of dependencies when plugin resources like `text!` are included, it fails to ignore the module id in the built file and tries to call the plugin (guess). 

removing the bundle from `exclude: []` solves it and it builds fine. 

also, editing built bundle file and replacing `text!` with `text=` works. `define('text=main.html',[],function () { return 'I am a teapot\n';});` builds fine with matching `define('baz',['require','text=./main.html','./foo','./bar'],function(require){`

I have a small repo that highlights this issue with grunt tasks to build both the bundle and the consumer file, if needed. could be that this is a problem with the requaire js plugin system or text! itself but i doubt it--appologies in advance and happy to move over if this is the case
